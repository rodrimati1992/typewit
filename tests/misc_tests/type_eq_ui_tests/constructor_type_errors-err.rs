use typewit::TypeEq;

const _: () = { TypeEq::<(), u8>::NEW; };

const fn new_assoc_const_with_differnt_types() -> TypeEq<i8, u8> {
    TypeEq::NEW
}

const fn new_function_with_differnt_types() -> TypeEq<i8, u8> {
    TypeEq::new()
}

const fn coercing_to_dyn_right() -> TypeEq<u8, dyn std::fmt::Debug> {
    TypeEq::new::<u8>()
}

const fn coercing_to_dyn_left() -> TypeEq<dyn std::fmt::Debug, u8> {
    TypeEq::new::<u8>()
}

const fn coercing_to_dyn_botht() -> TypeEq<dyn std::fmt::Debug, dyn std::fmt::Debug> {
    TypeEq::new::<u8>()
}


fn main(){}