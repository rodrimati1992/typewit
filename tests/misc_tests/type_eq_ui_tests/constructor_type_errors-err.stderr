error[E0599]: no associated item named `NEW` found for struct `TypeEq<(), u8>` in the current scope
 --> tests/misc_tests/type_eq_ui_tests/constructor_type_errors-err.rs:3:35
  |
3 | const _: () = { TypeEq::<(), u8>::NEW; };
  |                                   ^^^ associated item not found in `TypeEq<(), u8>`
  |
  = note: the associated item was found for
          - `TypeEq<T, T>`

error[E0308]: mismatched types
 --> tests/misc_tests/type_eq_ui_tests/constructor_type_errors-err.rs:6:5
  |
5 | const fn new_assoc_const_with_differnt_types() -> TypeEq<i8, u8> {
  |                                                   -------------- expected `TypeEq<i8, u8>` because of return type
6 |     TypeEq::NEW
  |     ^^^^^^^^^^^ expected `TypeEq<i8, u8>`, found `TypeEq<_, _>`
  |
  = note: expected struct `TypeEq<i8, u8>`
             found struct `TypeEq<_, _>`

error[E0308]: mismatched types
  --> tests/misc_tests/type_eq_ui_tests/constructor_type_errors-err.rs:10:5
   |
9  | const fn new_function_with_differnt_types() -> TypeEq<i8, u8> {
   |                                                -------------- expected `TypeEq<i8, u8>` because of return type
10 |     TypeEq::new()
   |     ^^^^^^^^^^^^^ expected `TypeEq<i8, u8>`, found `TypeEq<_, _>`
   |
   = note: expected struct `TypeEq<i8, u8>`
              found struct `TypeEq<_, _>`

error[E0308]: mismatched types
  --> tests/misc_tests/type_eq_ui_tests/constructor_type_errors-err.rs:14:5
   |
13 | const fn coercing_to_dyn_right() -> TypeEq<u8, dyn std::fmt::Debug> {
   |                                     ------------------------------- expected `TypeEq<u8, (dyn Debug + 'static)>` because of return type
14 |     TypeEq::new::<u8>()
   |     ^^^^^^^^^^^^^^^^^^^ expected `TypeEq<u8, dyn Debug>`, found `TypeEq<u8, u8>`
   |
   = note: expected struct `TypeEq<_, (dyn Debug + 'static)>`
              found struct `TypeEq<_, u8>`

error[E0308]: mismatched types
  --> tests/misc_tests/type_eq_ui_tests/constructor_type_errors-err.rs:18:5
   |
17 | const fn coercing_to_dyn_left() -> TypeEq<dyn std::fmt::Debug, u8> {
   |                                    ------------------------------- expected `TypeEq<(dyn Debug + 'static), u8>` because of return type
18 |     TypeEq::new::<u8>()
   |     ^^^^^^^^^^^^^^^^^^^ expected `TypeEq<dyn Debug, u8>`, found `TypeEq<u8, u8>`
   |
   = note: expected struct `TypeEq<(dyn Debug + 'static), _>`
              found struct `TypeEq<u8, _>`

error[E0308]: mismatched types
  --> tests/misc_tests/type_eq_ui_tests/constructor_type_errors-err.rs:22:5
   |
21 | const fn coercing_to_dyn_botht() -> TypeEq<dyn std::fmt::Debug, dyn std::fmt::Debug> {
   |                                     ------------------------------------------------ expected `TypeEq<(dyn Debug + 'static), (dyn Debug + 'static)>` because of return type
22 |     TypeEq::new::<u8>()
   |     ^^^^^^^^^^^^^^^^^^^ expected `TypeEq<dyn Debug, dyn Debug>`, found `TypeEq<u8, u8>`
   |
   = note: expected struct `TypeEq<(dyn Debug + 'static), (dyn Debug + 'static)>`
              found struct `TypeEq<u8, u8>`
